saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/RelyingPartyRegistration.java

```diff
@@ -288,17 +393,80 @@ public Builder localEntityIdTemplate(String template) {
 			return this;
 		}
 
+		/**
+		 * Constructs a RelyingPartyRegistration object based on the builder configurations
+		 * @return a RelyingPartyRegistration instance
+		 */
 		public RelyingPartyRegistration build() {
 			return new RelyingPartyRegistration(
-					remoteIdpEntityId,
 					registrationId,
 					assertionConsumerServiceUrlTemplate,
-					idpWebSsoUrl,
+					new ProviderDetails(
```

Comment:
Let's put this construction inside a `build()` method for `ProviderDetails`'s builder. It's a bit odd to have a builder, but then not have the builder perform the actual construction.

Reply:
Agreed.
c7937d150082ac8fd4a1ede9556876e10985d820